xbmc/Application.cpp

```diff
@@ -5010,12 +5010,36 @@ void CApplication::StopMusicScan()
     m_musicInfoScanner->Stop();
 }
 
-void CApplication::StartVideoCleanup(bool userInitiated /* = true */)
+void CApplication::StartVideoCleanup(bool userInitiated /* = true */,
+                                     const std::string& content /* = "" */)
 {
   if (userInitiated && CVideoLibraryQueue::GetInstance().IsRunning())
     return;
 
   std::set<int> paths;
+  if (!content.empty())
```

Comment:
Reverse check to avoid some nesting, same for next if

Reply:
i don't get it.

Reply:
if you do `if (content.empty) return paths;` and same with next if there's too levels less of nesting/indentation and easier to read. But as I said it's a matter of taste so I'm just mentioning it

Reply:
I an not returning, it i am sending the array into the db clean after the if nest.

Reply:
right, my example above should've just been `if (paths.empty()) return;` and `if (!(db.Open() && db.GetPaths(contentPaths)) return;`

Reply:
Which still would be wrong. If no content is specified we clean all content type and in that case we pass an empty vector.

Reply:
oh right. I see it now. Sorry about that.